src/components/dashboard/main-content/course-enrollments/course-cards/CompletedCourseCard.jsx

```diff
@@ -1,36 +1,143 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
 import { getAuthenticatedUser } from '@edx/frontend-platform/auth';
+import { sendTrackEvent } from '@edx/frontend-platform/analytics';
+import { courseEnded } from '../../../../../utils/common';
+import ContinueLearningButton from './ContinueLearningButton';
+
+import {
+  updateCourseRunStatus,
+  updateIsUnarchiveCourseSuccess,
+} from '../data/actions';
 
 import BaseCourseCard from './BaseCourseCard';
 
+import { UnarchiveModal } from './unarchive-modal';
+
 import CertificateImg from './images/edx-verified-mini-cert.png';
 
-const CompletedCourseCard = props => {
+const CompletedCourseCard = (props) => {
   const user = getAuthenticatedUser();
   const { username } = user;
-  return (
-    <BaseCourseCard type="completed" hasViewCertificateLink={false} {...props}>
-      {props.linkToCertificate ? (
-        <div className="d-flex mb-3">
-          <div className="mr-3">
-            <img src={CertificateImg} alt="verified certificate preview" />
-          </div>
-          <div className="d-flex align-items-center">
-            <p className="lead mb-0 font-weight-normal">
-              View your certificate on
-              {' '}
-              <a className="text-underline" href={`${process.env.LMS_BASE_URL}/u/${username}`}>your profile →</a>
-            </p>
+  const {
+    markedDone,
+    title,
+    linkToCourse,
+    courseRunId,
+    modifyCourseRunStatus,
+    modifyIsUnarchiveCourseStatus,
+    endDate,
+  } = props;
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+
+  const handleUnarchiveOnClose = () => {
+    setIsModalOpen(false);
+    sendTrackEvent('edx.learner_portal.course.unarchive.modal.closed', {
+      course_run_id: courseRunId,
+    });
+  };
+
+  const handleUnarchiveOnSuccess = ({ response, resetModalState }) => {
+    sendTrackEvent('edx.learner_portal.course.unarchive.saved', {
+      course_run_id: courseRunId,
+    });
+    setIsModalOpen(false);
+    resetModalState();
+    modifyCourseRunStatus({
+      status: response.courseRunStatus,
+      courseId: response.courseRunId,
+      markedDone: response.markedDone.toLowerCase() === 'true',
```

Comment:
Does the API return `markedDone` as a string? If so, wondering if we should fix the API to return an actual boolean so we don't need to convert the string to a boolean.

Reply:
yup it does right "False" is what I got. So yes I think we should task out the fixing of api and subsequent changing of code here

Reply:
tasked out to https://openedx.atlassian.net/browse/ENT-3047